api/client/client.go

```diff
@@ -828,23 +828,21 @@ func (c *Client) GetKubeServices(ctx context.Context) ([]types.Server, error) {
 
 // GetApplicationServers returns all registered application servers.
 func (c *Client) GetApplicationServers(ctx context.Context, namespace string) ([]types.AppServer, error) {
-	resp, err := c.grpc.GetApplicationServers(ctx, &proto.GetApplicationServersRequest{
-		Namespace: namespace,
-	}, c.callOpts...)
+	resources, err := c.ListAllResources(ctx, namespace, proto.ResourceType_RESOURCE_TYPE_APPLICATION_SERVER)
 	if err != nil {
-		if trace.IsNotImplemented(trail.FromGRPC(err)) {
-			servers, err := c.getApplicationServersFallback(ctx, namespace)
-			if err != nil {
-				return nil, trace.Wrap(err)
-			}
-			return servers, nil
-		}
-		return nil, trail.FromGRPC(err)
+		return nil, trace.Wrap(err)
```

Comment:
Don't we need some kind of fallback here in cases if ListResources is not implemented  in auth service ?

Reply:
Yeah, I added it and covered the necessity of returning legacy application servers before 9.0 (made on the `GetApplicationServers` function).